receiver/prometheusreceiver/config.go

```diff
@@ -110,17 +113,30 @@ func (cfg *Config) Unmarshal(componentParser *configparser.Parser) error {
 		return fmt.Errorf("prometheus receiver failed to parse config: %s", err)
 	}
 
-	// Unmarshal prometheus's config values. Since prometheus uses `yaml` tags, so use `yaml`.
-	promCfg, err := componentParser.Sub(prometheusConfigKey)
-	if err != nil || len(promCfg.ToStringMap()) == 0 {
-		return err
-	}
-	out, err := yaml.Marshal(promCfg.ToStringMap())
-	if err != nil {
-		return fmt.Errorf("prometheus receiver failed to marshal config to yaml: %s", err)
+	var yamlBlob []byte
+
+	// TODO: Ensure that only 1 of "config_file" or the configuration exist.
+
+	switch cfg.File == "" {
```

Comment:
An `if`/`else` statement would be more idiomatic here.

Reply:
I don't agree on your definition of "idiomatic" here, I'd even argue that the if/else makes reading the code harder because of inability to display a newline and remove the braces before the next case; however I've added it. Please take a look and thank you!